---
title: "Seminar 3"
author: "Julia A"
date: "January 22, 2018"
output: github_document
---

# Install The Following
```{r}
library(knitr)
library(GEOquery)
library(biomaRt)
library(tidyverse)
library(data.table)
library(reshape2)
```

## Part 1
```{r}
gds <- getGEO("GDS507")
str(gds)
meta_data <- data.frame(Sample = gds@dataTable@columns$sample, disease = gds@dataTable@columns$disease.state) #we grab this info from the appropriate slots above. 
data <- gds@dataTable@table
head(data)
dim(data)
apply(data[,-c(1, 2)], 2, median)
melted_data <- melt(data, id.vars = c("ID_REF", "IDENTIFIER"), var = "Sample")
head(melted_data)

#mean gene expression per sample
melted_data %>% 
    group_by(Sample) %>% 
    summarize(mean = mean(value))

#mean of each probe's expression
new_melted_data <- melted_data %>% 
    group_by(Sample, IDENTIFIER) %>% 
    summarize(Count = mean(value))

#function that takes in data frame, and outputs same data frame with associated chromosome annotations.
identify_gene_names <- function(df){
    names(df) <- c("Sample", "hgnc_symbol", "Count")
    names <- getBM(attributes=c("hgnc_symbol", "chromosome_name") , filters= "hgnc_symbol", values = df$hgnc_symbol, mart = human)
    left_join(df, names, by = "hgnc_symbol")
}

data_with_chromosome <- read.csv("https://raw.githubusercontent.com/STAT540-UBC/STAT540-UBC.github.io/master/seminars/seminars_winter_2017/Seminar3/biomart_output.csv", header=TRUE)

full_data <- left_join(data_with_chromosome, meta_data, by = "Sample")

full_data %>% 
    group_by(disease) %>% 
    filter(chromosome_name == "X") %>% 
    summarize(mean = mean(Count))

#choose random number between 1 and however many genes we have. 
sample_to_choose <- sample(1:length(unique(full_data$hgnc_symbol)), size = 100)
#choose genes that correspond to those numbers in a list of genes. 
names_to_choose <- as.character(unique(full_data$hgnc_symbol)[sample_to_choose])

full_data %>% 
    filter(hgnc_symbol %in% names_to_choose) %>% 
    group_by(Sample) %>% mutate(relExp = Count/sum(Count)) %>%
    ggplot(aes(x = as.factor(chromosome_name), y = relExp)) + geom_point()

full_data %>% 
    group_by(hgnc_symbol) %>% 
   summarize( pvalue = t.test(Count ~ disease)$p.value) %>%
   ggplot(aes(x = pvalue)) + geom_density() 
 
data_05 <- full_data %>% 
    group_by(hgnc_symbol) %>% 
    summarize(pvalue = t.test(Count ~ disease)$p.value) %>% 
  filter(pvalue < 0.05)

data_05
```

